packages/block-editor/src/components/use-moving-animation/index.js

```diff
@@ -60,14 +60,35 @@ function useMovingAnimation(
 		0
 	);
 	const [ finishedAnimation, endAnimation ] = useReducer( counterReducer, 0 );
-	const [ transform, setTransform ] = useState( {
-		x: 0,
-		y: 0,
-		clientTop: 0,
-	} );
+	const [ transform, setTransform ] = useState( { x: 0, y: 0 } );
+	const previous = useMemo(
+		() => ( ref.current ? getAbsolutePosition( ref.current ) : null ),
+		[ triggerAnimationOnChange ]
+	);
+
+	// Calculate the previous position of the block relative to the viewport and
+	// return a function to maintain that position by scrolling.
+	const preserveScrollPosition = useMemo( () => {
```

Comment:
All scroll logic is now nicely contained.

Reply:
I like it 👍 We could return "noop" when we don't need to scroll to avoid the test done later.